.github/workflows/integration-tests.yaml

```diff
@@ -54,6 +54,11 @@ jobs:
       - name: Show Cargo.toml After Update
         run: cat rust/Cargo.toml  # Correct path to the Cargo.toml file
 
+      # Ensure Cargo.lock is updated with the latest dependencies
+      - name: Update Dependencies
+        run: cargo update
```

Comment:
this seems very risky... how to prevent new dependency from breaking changes..

Reply:
hmm yeah. but this doesn't override our cargo.lock, so it shouldn't be as risky. 

maybe I need to understand why it's building with the previous integration tests first.

Reply:
oh i see i read it in the wrong way..
i thought it's upgrade lol

Reply:
ohhhhh haha tyty!!